cmd/clusterctl/api/v1alpha3/provider_type.go

```diff
@@ -24,15 +24,20 @@ import (
 // +kubebuilder:resource:path=providers,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".provider"
 // +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".type"
 // +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".version"
 // +kubebuilder:printcolumn:name="Watch Namespace",type="string",JSONPath=".watchedNamespace"
 
-// Provider is the Schema for the providers API
+// Provider defines an entry in the provider inventory.
 type Provider struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// Provider indicates the name of the provider.
+	// +optional
+	Provider string `json:"provider,omitempty"`
```

Comment:
```suggestion
	Name string `json:"name,omitempty"`
```

Suggest changing this to name since the type is Provider already. I noticed later on that we have things like `provider.Provider.Provider` which is a little hard to read.

Reply:
Doesn't metadata already have a name?